File: QuickExtractingLib2/ConvMatrix.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QuickExtractingLib2
{
    using System;

    internal class ConvMatrix : ICloneable
    {
        public int TopLeft = 0;
        public int TopMid = 0;
        public int TopRight = 0;
        public int MidLeft = 0;
        public int Pixel = 1;
        public int MidRight = 0;
        public int BottomLeft = 0;
        public int BottomMid = 0;
        public int BottomRight = 0;
        public int Factor = 1;
        public int Offset = 0;

        public void SetAll(int nVal)
        {
            TopLeft = InlineAssignHelper(ref TopMid, InlineAssignHelper(ref TopRight, InlineAssignHelper(ref MidLeft, InlineAssignHelper(ref Pixel, InlineAssignHelper(ref MidRight, InlineAssignHelper(ref BottomLeft, InlineAssignHelper(ref BottomMid, InlineAssignHelper(ref BottomRight, nVal))))))));
        }
        private static T InlineAssignHelper<T>(ref T target, T value)
        {
            target = value;
            return value;
        }

        public object Clone()
        {
            ConvMatrix cm = new ConvMatrix();
            cm.BottomLeft = this.BottomLeft;
            cm.BottomMid = this.BottomMid;
            cm.BottomRight = this.BottomRight;
            cm.Factor = this.Factor;
            cm.MidLeft = this.MidLeft;
            cm.MidRight = this.MidRight;
            cm.Offset = this.Offset;
            cm.Pixel = this.Pixel;
            cm.TopLeft = this.TopLeft;
            cm.TopMid = this.TopMid;
            cm.TopRight = this.TopRight;
            return cm;
        }
    }

}

File: ConvolutionLib/ConvMatrix.vb
Option Strict On

Public Class ConvMatrix
    Implements ICloneable

    Public TopLeft As Integer = 0, TopMid As Integer = 0, TopRight As Integer = 0
    Public MidLeft As Integer = 0, Pixel As Integer = 1, MidRight As Integer = 0
    Public BottomLeft As Integer = 0, BottomMid As Integer = 0, BottomRight As Integer = 0

    Public Factor As Integer = 1
    Public Offset As Integer = 0

    Public Sub SetAll(nVal As Integer)
        TopLeft = InlineAssignHelper(TopMid, InlineAssignHelper(TopRight, InlineAssignHelper(MidLeft, InlineAssignHelper(Pixel, InlineAssignHelper(MidRight, InlineAssignHelper(BottomLeft, InlineAssignHelper(BottomMid, InlineAssignHelper(BottomRight, nVal))))))))
    End Sub
    Private Shared Function InlineAssignHelper(Of T)(ByRef target As T, value As T) As T
        target = value
        Return value
    End Function

    Public Function Clone() As Object Implements ICloneable.Clone
        Dim cm As New ConvMatrix()
        cm.BottomLeft = Me.BottomLeft
        cm.BottomMid = Me.BottomMid
        cm.BottomRight = Me.BottomRight
        cm.Factor = Me.Factor
        cm.MidLeft = Me.MidLeft
        cm.MidRight = Me.MidRight
        cm.Offset = Me.Offset
        cm.Pixel = Me.Pixel
        cm.TopLeft = Me.TopLeft
        cm.TopMid = Me.TopMid
        cm.TopRight = Me.TopRight
        Return cm
    End Function
End Class
